app/services/find_empty_trainees.rb

```diff
@@ -4,72 +4,96 @@
 class FindEmptyTrainees
   include ServicePattern
 
-  EXCLUDED_FIELDS = %w[
-    created_at
-    updated_at
-    dttp_update_sha
-    progress
-    slug
-    training_route
-    id
-    provider_id
-    ebacc
+  FIELDS_TO_CHECK = %w[
+    first_names
+    last_name
+    date_of_birth
+    address_line_one
+    address_line_two
+    town_city
+    postcode
+    email
+    middle_names
+    international_address
+    trainees.locale_code
+    gender
+    diversity_disclosure
+    ethnic_group
+    ethnic_background
+    additional_ethnic_background
+    disability_disclosure
+    course_subject_one
+    course_start_date
+    outcome_date
+    course_end_date
+    trn
+    submitted_for_trn_at
+    withdraw_reason
+    withdraw_date
+    additional_withdraw_reason
+    defer_date
+    recommended_for_award_at
+    commencement_date
+    reinstate_date
+    lead_school_id
+    employing_school_id
+    apply_application_id
+    course_min_age
+    course_max_age
+    course_code
+    course_subject_two
+    course_subject_three
+    awarded_at
+    applying_for_bursary
+    training_initiative
+    bursary_tier
+    trainees.study_mode
+    region
+    course_education_phase
+    applying_for_scholarship
+  ].freeze
+
+  EARLY_YEARS_FIELDS_TO_EXCLUDE = %w[
+    course_subject_one
+    course_min_age
+    course_max_age
   ].freeze
 
   class FieldsDoNotExistError < StandardError; end
 
   attr_reader :trainees, :ids_only, :forms
 
   def initialize(trainees: Trainee.all, ids_only: false)
-    raise FieldsDoNotExistError unless exluded_fields_exist?
-
     @trainees = trainees
     @ids_only = ids_only
   end
 
   def call
-    draft_trainees.filter_map do |trainee|
-      if no_form_data_for?(trainee)
-        ids_only ? trainee.id : trainee
-      end
-    end
+    ids_only ? draft_trainees.pluck(:id) : draft_trainees
   end
 
 private
 
-  def exluded_fields_exist?
-    EXCLUDED_FIELDS.all? do |field|
-      Trainee.column_names.include?(field)
-    end
-  end
-
   def draft_trainees
```

Comment:
Could we be more explicit naming this? `empty_draft_trainees` or `blank...` or something like that?